helm/vitess/templates/_vttablet.tpl

```diff
@@ -114,7 +115,9 @@ spec:
         - name: vt
           emptyDir: {}
 {{ include "backup-volume" $config.backup | indent 8 }}
-{{ include "user-config-volume" $defaultVttablet.extraMyCnf | indent 8 }}
+{{ include "user-config-volume" (.extraMyCnf | default $defaultVttablet.extraMyCnf) | indent 8 }}
```

Comment:
This seems a little confusing since right below, secrets and default secrets are both applied, but here, the default `extraMyCnf` is only applied if `extraMyCnf` is empty. This holds for a lot of the config/secret/volume setup in this PR.

Maybe we should have separate keys in the `values.yaml`?
- `mergeExtraMyCnf`
- `replaceExtraMyCnf`

It seems verbose, so I don't necessarily love the idea. I'm open to suggestions about how to make it clear whether how the defaults are going to be combined with any unique defaults per shard/tablet/etc.

Reply:
The behavior of `extraMyCnf` seems to be inline with the rest of the values, where e.g. in the `vttablet` section the default is specified and can be overridden on a per topology basis.

The bigger issue is how the secrets differ from this default behavior. It would probably be cleaner if secrets override their defaults as well.

Reply:
I changed the behavior of secrets to fit the rest of the `values.yaml`: https://github.com/vitessio/vitess/pull/4333/commits/a09b0cd7f8c4e66db7b53dadede6e60194750c9f

Reply:
LGTM. We can revisit having some global options that merge with overridden options if the need arises.